backends/dpdk/dpdkHelpers.cpp

```diff
@@ -630,9 +630,17 @@ bool ConvertStatementToDpdk::preorder(const IR::MethodCallStatement *s) {
             } else if (boolLiteral->value == true) {
                 return false;
             }
-            auto tmp = new IR::PathExpression(
-                                   IR::ID("m.psa_ingress_input_metadata_parser_error"));
-            add_instr(new IR::DpdkMovStatement(tmp, new IR::Constant(error_id)));
+            IR::PathExpression *error_meta_path;
+            if (structure->p4arch == "psa") {
```

Comment:
maybe it's time to add some helper functions isPSA(), isPNA()

Reply:
I added methods to DpdkProgramStructure.